### PR TITLE
Homepage collection don't assign unpublished

### LINKED ISSUE #2562: Assigning an unpublished collection as a homepage collection
It is possible to assign the unpublished collection as a homepage collection and display it publicly.

### Steps to reproduce the problem
1. Create a collection of products but don't publish it.
2. Go to the collection list view and in the "Homepage collection" card choose the unpublished collection. Submit the form.
3. Go to the storefront's homepage. The featured products section at the bottom shows the unpublished collection.

### What I expected to happen
A validation error in the "Hompage collection" could be raised when trying to assign an unpublished collection.

### PR DESCRIPTION
**Pull request by oldPadavan**
I want to merge this change because...

closes #2562 

### Screenshots

2018-08-21-145423_1920x951_scrot

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).

### CODE REVIEW

File: saleor/dashboard/collection/forms.py [hunk 1]
```diff
@@ -45,3 +45,12 @@ class AssignHomepageCollectionForm(forms.ModelForm):
     class Meta:
         model = SiteSettings
         fields = ('homepage_collection',)
+
+    def clean(self):
```
Comment (Pacu2) [inline saleor/dashboard/collection/forms.py]: I think this should be placed in `clean_homepage_collection` function, what's more, we could change field's queryset to get only collections that are published.

File: saleor/dashboard/collection/forms.py [hunk 2]
```diff
@@ -45,3 +45,8 @@ class AssignHomepageCollectionForm(forms.ModelForm):
     class Meta:
         model = SiteSettings
         fields = ('homepage_collection',)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields['homepage_collection'].queryset = Collection.objects.\
```
Comment (Pacu2) [inline saleor/dashboard/collection/forms.py]: I think we could break the line after `queryset =` to make it a little bit prettier

File: templates/dashboard/collection/detail.html [hunk 3]
```diff
@@ -89,7 +89,8 @@
                   <form id="toggle-publish-form" action="{% url 'dashboard:collection-publish' pk=collection.pk %}">
                     <div class="switch" id="collection-is-published">
                       <label>
-                        <input id="toggle-publish-switch" type="checkbox" {% if collection.is_published %} checked {% endif %}>
+                        <input id="toggle-publish-switch" type="checkbox" {% if collection.is_published %} checked {% endif %}
+                        {% if site.settings.homepage_collection == collection and collection.is_published %} disabled {% endif %} >
```
Comment (maarcingebala) [inline templates/dashboard/collection/detail.html]: `site.settings.homepage_collection == collection and collection.is_published` - this check could be done in the view function with the result passed to the template through context. First, it is a good practice to perform database queries in the view where we can ensure that necessary prefetching is made. Second, this line is duplicated in the template.

**Comment by maarcingebala** [comment]
Thanks!

**Pull request merged**